test/addons/load-long-path/test.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+const common = require('../../common');
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+
+common.refreshTmpDir();
+
+// make a path that is more than 260 chars long.
+// Any given folder cannot have a name longer than 260 characters,
+// so create 10 nested folders each with 30 character long names.
+var addonDestinationDir = path.resolve(common.tmpDir);
+
+for (var i = 0; i < 10; i++) {
+  addonDestinationDir = path.join(addonDestinationDir, 'x'.repeat(30));
+  fs.mkdirSync(addonDestinationDir);
+}
+
+const addonPath = path.join(__dirname, 'build', 'Release', 'binding.node');
+const addonDestinationPath = path.join(addonDestinationDir, 'binding.node');
+
+// Copy binary to long path destination
+var contents = fs.readFileSync(addonPath);
+fs.writeFileSync(addonDestinationPath, contents);
+
+// Attempt to load at long path destination
+var addon = require(addonDestinationPath);
+assert(addon != null);
```

Comment:
@thefourtheye  I believe I just copied that line from another test doing something similar. However, I think its most accurate to say that `addon` should be neither `null` or `undefined`. So yes it is intentional.

Also, the error case before this PR actually throws an error instead of returning `null` or `undefined`. So it's mostly just a sanity check. Same with the following line which actually calls the `addon.hello()` function, another sanity check.